docs/index.bs

```diff
@@ -3025,11 +3025,12 @@ spec: rfc7231; urlPrefix: https://tools.ietf.org/html/rfc7231
       <li>Let <var>httpsState</var> be "<code>none</code>".</li>
       <li>Let <var>referrerPolicy</var> be the empty string.</li>
       <li>Switching on <var>job</var>'s <a>worker type</a>, run these substeps with the following options:
+        <!-- TODO: reorganize algorithm so that the worker environment is created before fetching happens -->
         <dl>
         <dt><em>"<code>classic</code>"</em></dt>
-        <dd><p><a>Fetch a classic worker script</a> given <var>job</var>’s <a lt="URL serializer">serialized</a> <a href="#dfn-job-script-url">script url</a>, <var>job</var>’s <a href="#dfn-job-client">client</a>, and "<code>serviceworker</code>".</p></dd>
+        <dd><p><a>Fetch a classic worker script</a> given <var>job</var>’s <a lt="URL serializer">serialized</a> <a href="#dfn-job-script-url">script url</a>, <var>job</var>’s <a href="#dfn-job-client">client</a>, "<code>serviceworker</code>", and the to-be-created <a>environment settings object</a> for this service worker.</p></dd>
```

Comment:
I'd prefer to fix this in a separate issue. It seems really hard to me to fix. You can't just mutate a script's settings object slot like that; the settings object determines how the script is parsed and some other important things, which happens during fetching. You can't pass null in that case as otherwise it's impossible to parse the script.